lxd/storage_pools.go

```diff
@@ -557,6 +557,23 @@ func storagePoolDelete(d *Daemon, r *http.Request) Response {
 		return EmptySyncResponse
 	}
 
+	volumeNames, err := d.cluster.StoragePoolVolumesGetNames(poolID)
```

Comment:
Hm, what happens if there are non image volumes still existent? Would this still delete all images and then fail?

Reply:
If there are non-image-volumes existent, this code won't be reached but fail when calling `storagePoolDeleteCheckPreconditions`.

Reply:
Ok